File: day02/src/test/java/de/havox_design/aoc2015/day02/Day02Test.java
package de.havox_design.aoc2015.day02;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

import java.util.stream.Stream;

class Day02Test {

    @SuppressWarnings("java:S2699")
    @Test
    void testMainClass() {
        MainClass.main(new String[0]);
    }

    @ParameterizedTest
    @MethodSource("getDataForPart1")
    void testPart1(String fileName, int expectedSquareFoots) {
        Assertions.assertEquals(expectedSquareFoots, WrappingPaper.solvePart1(fileName));
    }

    private static Stream<Arguments> getDataForPart1() {
        return Stream.of(
                Arguments.of("part1sample1.txt", 58),
                Arguments.of("part1sample2.txt", 43),
                Arguments.of("part1sample3.txt", 101)
        );
    }
}

File: aoc2022/src/test/kotlin/de/havox_design/aoc2022/day02/Day02Test.kt
package de.havox_design.aoc2022.day02

import de.havox_design.aoc2022.MainClass
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.CsvSource

class Day02Test {
    @Test
    fun testMainClass() {
        MainClass.main(arrayOf("day02"))
    }

    @ParameterizedTest
    @CsvSource(
        "de/havox_design/aoc2022/day02/day02Rock_rock.txt, 4",
        "de/havox_design/aoc2022/day02/day02Rock_paper.txt, 8",
        "de/havox_design/aoc2022/day02/day02Rock_scissors.txt, 3",
        "de/havox_design/aoc2022/day02/day02Paper_rock.txt, 1",
        "de/havox_design/aoc2022/day02/day02Paper_paper.txt, 5",
        "de/havox_design/aoc2022/day02/day02Paper_scissors.txt, 9",
        "de/havox_design/aoc2022/day02/day02Scissors_rock.txt, 7",
        "de/havox_design/aoc2022/day02/day02Scissors_paper.txt, 3",
        "de/havox_design/aoc2022/day02/day02Scissors_scissors.txt, 6",
        "de/havox_design/aoc2022/day02/day02Sample.txt, 15"
    )
    fun testPlaybooks(filename: String, expectedScore: Int) {
        RockPaperScissorsGame(filename)
            .getResultForGuide()
            .shouldBe(expectedScore)
    }

    @ParameterizedTest
    @CsvSource(
        "de/havox_design/aoc2022/day02/day02Rock_win.txt, 8",
        "de/havox_design/aoc2022/day02/day02Rock_draw.txt, 4",
        "de/havox_design/aoc2022/day02/day02Rock_loss.txt, 3",
        "de/havox_design/aoc2022/day02/day02Paper_win.txt, 9",
        "de/havox_design/aoc2022/day02/day02Paper_draw.txt, 5",
        "de/havox_design/aoc2022/day02/day02Paper_loss.txt, 1",
        "de/havox_design/aoc2022/day02/day02Scissors_win.txt, 7",
        "de/havox_design/aoc2022/day02/day02Scissors_draw.txt, 6",
        "de/havox_design/aoc2022/day02/day02Scissors_loss.txt, 2",
        "de/havox_design/aoc2022/day02/day02Sample.txt, 12"
    )
    fun testGameByExpectedResults(filename: String, expectedScore: Int) {
        RockPaperScissorsGame(filename)
            .getResultForExpectedResult()
            .shouldBe(expectedScore)
    }
}

private fun Int.shouldBe(expectation: Int) = Assertions.assertEquals(expectation, this)
